app/tasks.py

```diff
@@ -1,25 +1,88 @@
 import json
-from app import celery
-from app.base.file_handler import property_image_handler
+import os
+from PIL import Image
+from app import celery, s3
+from app.base.utils import update_property_images
+from config import S3_BUCKET_CONFIG
+
+bucket = S3_BUCKET_CONFIG["S3_BUCKET"]
+s3_prop_image_dir = S3_BUCKET_CONFIG["PROP_ASSETS"]
+s3_temp_dir = S3_BUCKET_CONFIG["TEMP_DIR"]
+s3_user_image_dir = S3_BUCKET_CONFIG["USER_ASSETS"]
 
 
 @celery.task()
-def save_property_data(temp_folder, form_data):
-    from app.base.models import Property
+def process_images(filename, tmp_dir, dir_to_save):
+    """
+    This function processes the image files, using the PIL image library. The filename argument is
+    used to create writable binary object, the object is overwritten by the data of the image downloaded
+    from s3 using boto3 client. Then the image processed using PIL is uploaded back to s3 using boto3 client.
+    The name of this object the filename argument. The image is deleted in the temp dir for cleanup.
+    """
+    try:
+        img_obj = open(filename, "wb")
+        s3.download_fileobj(bucket, f"{s3_temp_dir + tmp_dir}{filename}", img_obj)
+        pil_image_obj = Image.open(filename)
+        pil_image_obj.thumbnail((800, 800))
+        pil_image_obj.save(filename)
 
-    prop_images_dir, images_list = property_image_handler(temp_folder)
-    img_list_to_json = json.dumps(images_list)
-    prop_images = {"photos": img_list_to_json, "image_folder": prop_images_dir}
-    form_data.update(prop_images)
-    Property.add_property(form_data)
-    return "Property Created"
+        with open(filename, "rb") as data_obj:
+            s3.upload_fileobj(
+                data_obj,
+                bucket,
+                f"{s3_prop_image_dir}{dir_to_save}{filename}",
+                ExtraArgs={
+                    "ACL": "public-read",
+                    "ContentType": "image/jpeg"
+                }
+            )
+        os.remove(filename)
+        delete_img_objs(bucket, s3_temp_dir + tmp_dir, file_obj=filename)
+    except Exception as e:
+        return e
```

Comment:
I thought this logic was to be implemented in one place other rather than duplicating it accross utility functions and celery worker

the best is to let the worker do this or let the worker trigger the utility function which takes care of this 

that way its easy to maintain in case we updated the bucket to be private in the near future and the code needs updating

or the alternative would be to let only the worker know about all the s3 uploads and all 
especially when it starts getting temp images to process directly  from redis

Reply:
Yeah, I saw some duplication there, but I have resolved it